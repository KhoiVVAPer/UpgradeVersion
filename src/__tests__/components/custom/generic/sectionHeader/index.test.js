import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SectionHeader from '../../../../../components/custom/generic/sectionHeader';

const setUp = (props) => {
  const component = shallow(<SectionHeader {...props} />);
  return component;
};

describe('components > custom > elements > SectionHeader', () => {
  let component;
  let props = {
    heading: 'Heading'
  };
  beforeEach(() => {
    component = setUp(props);
    return component;
  });

  it('Check snapshot', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders component', () => {
    const input = component.find('View').length;
    const output = 1;
    expect(input).toEqual(output);
  });

  it('Renders heading', () => {
    const input = component.find('Text').length;
    const output = 1;
    expect(input).toEqual(output);
  });

  it('Renders heading with marking feature', () => {
    props = {
      ...props,
      marking: true
    };
    component = setUp(props);
    const input = component.find('HtmlParser').length;
    const output = 1;
    expect(input).toEqual(output);
  });
});
