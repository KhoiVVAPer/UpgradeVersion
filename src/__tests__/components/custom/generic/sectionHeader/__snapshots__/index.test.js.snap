// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > custom > elements > SectionHeader Check snapshot 1`] = `
<react-native-mock>
  <react-native-mock>
    HEADING
  </react-native-mock>
</react-native-mock>
`;
